src/transformers/modeling_utils.py

```diff
@@ -4002,7 +4002,7 @@ def from_pretrained(
                 pass
 
         # Dispatch model with hooks on all devices if necessary
-        if device_map is not None:
+        if device_map is not None and device_map != {'': torch.device(type='cpu')}:
```

Comment:
We shouldn't add that here. It should be taken care of in the dispatch function itself. If you don't want to modify the `dispatch_model` (requires to update accelerate), then you can the bnb +cpu condition just above of where we use `dispatch_model`.